nvflare/private/fed/client/admin.py

```diff
@@ -112,47 +115,79 @@ def _dispatch_request(
-                        elif not authorized:
-                            reply = error_reply("not authorized")
-                    else:
-                        reply = error_reply("requires authz but missing admin command")
-
-                if not reply:
-                    reply = processor.process(req, self.app_ctx)
-                    if reply is None:
-                        # simply ack
-                        reply = ok_reply()
-                    else:
-                        if not isinstance(reply, Message):
-                            raise RuntimeError(f"processor for topic {topic} failed to produce valid reply")
-            except Exception as e:
-                secure_log_traceback()
-                reply = error_reply(f"exception_occurred: {secure_format_exception(e)}")
+            with self.app_ctx.new_context() as fl_ctx:
+                peer_props = req.get_header(ReservedHeaderKey.PEER_PROPS)
+                if peer_props:
+                    peer_ctx = FLContext()
+                    peer_ctx.set_public_props(peer_props)
+                    fl_ctx.set_peer_context(peer_ctx)
+
+                try:
+                    reply = None
+
+                    # see whether pre-authorization is needed
+                    authz_flag = req.get_header(RequestHeader.REQUIRE_AUTHZ)
+                    require_authz = authz_flag == "true"
+                    if require_authz:
+                        # authorize this command!
+                        cmd = req.get_header(RequestHeader.ADMIN_COMMAND, None)
+                        if cmd:
+                            user = Person(
+                                name=req.get_header(RequestHeader.USER_NAME, ""),
+                                org=req.get_header(RequestHeader.USER_ORG, ""),
+                                role=req.get_header(RequestHeader.USER_ROLE, ""),
+                            )
+                            submitter = Person(
+                                name=req.get_header(RequestHeader.SUBMITTER_NAME, ""),
+                                org=req.get_header(RequestHeader.SUBMITTER_ORG, ""),
+                                role=req.get_header(RequestHeader.SUBMITTER_ROLE, ""),
+                            )
+                            custom_data = req.get_header(RequestHeader.CUSTOM_PROPS)
+                            if custom_data:
+                                fl_ctx.set_prop(FLContextKey.CUSTOM_PROPS, custom_data)
+
+                            authz_ctx = AuthzContext(user=user, submitter=submitter, right=cmd)
+                            authorized, err = AuthorizationService.authorize(authz_ctx)
+                            if err:
+                                reply = error_reply(err)
+                            elif not authorized:
+                                reply = error_reply("not authorized")
+
+                            site_security_filter = SiteSecurityFilter()
```

Comment:
Not sure about this. A filter is supposed to be used at the Admin Server side to filter command. But this is client side.

Reply:
Changed to not implement Filter.